### PR TITLE
YD-195 Include device time in app activity report

### PR DESCRIPTION
**Pull request by Bert-R**
Based on this, the time stamps of the reported app activity events are corrected. Along with this change, the API is added the the Swagger spec. This was somehow forgotten.

### CODE REVIEW

**Comment by aristotelos** [comment]
One test failing:  nu.yona.server.analysis.service.AnalysisEngineServiceTests.crossDayActivity

**Comment by Bert-R** [comment]
Yes, I noticed it yesterday evening, after creating the PR. It seems to be a timing issue that never occurs for me locally. I'll look into it.

File: appservice/src/main/resources/static/swagger/swagger-spec.yaml [hunk 1]
```diff
@@ -406,6 +400,69 @@ paths:
...
+      description: Reports the usage of apps on the device
+      parameters:
+        - in: body
+          name: body
+          description: The app activity report
+          required: true
+          schema:
+            type: object
+            required:
+              - newDeviceRequestPassword
```
Comment (aristotelos) [inline appservice/src/main/resources/static/swagger/swagger-spec.yaml]: Copy-paste error

File: core/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java [hunk 2]
```diff
@@ -45,17 +46,37 @@
...
+	private Duration determineDeviceTimeOffset(AppActivityDTO appActivities)
+	{
+		Duration offset = Duration.between(ZonedDateTime.now(), appActivities.getDeviceDateTime());
+		return (offset.abs().compareTo(Duration.ofSeconds(10)) > 0) ? offset : Duration.ZERO; // Ignore if less than 10 seconds
+	}
+
+	private ActivityPayload createActivityPayload(Duration deviceTimeOffset, AppActivityDTO.Activity appActivity)
+	{
+		Date correctedStartTime = correctTime(deviceTimeOffset, appActivity.getStartTime());
+		Date correctedendTime = correctTime(deviceTimeOffset, appActivity.getEndTime());
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java]: `correctedEndTime`

**Comment by aristotelos** [comment]
Looks great, found a few typos

**Pull request merged**